authentication/src/main/java/io/camunda/authentication/config/WebSecurityConfig.java

```diff
@@ -130,31 +136,66 @@ private void genericSuccessHandler(
   @Profile("auth-basic")
   @Order(1)
   public SecurityFilterChain httpBasicAuthSecurityFilterChain(
-      final HttpSecurity httpSecurity, final AuthFailureHandler authFailureHandler)
+      final HttpSecurity httpSecurity,
+      final AuthFailureHandler authFailureHandler,
+      final SecurityConfiguration securityConfiguration)
       throws Exception {
-    LOG.info("HTTP Basic authentication is enabled");
-    // Require valid credentials when a basic auth header is present.
+    final boolean isEnabled =
+        securityConfiguration.getAuthentication().getBasic().isHttpBasicAuthEnabled();
+    LOG.info("HTTP Basic authentication is {}", isEnabled ? "enabled" : "disabled");
+    if (!isEnabled) {
+      return null;
+    }
+    final boolean isApiUnprotected =
+        securityConfiguration.getAuthentication().getBasic().isApiDangerouslyUnprotected();
     return baseHttpSecurity(
-            withSecurityMatcher(httpSecurity, WebSecurityConfig::isBasicAuthRequest),
-            authFailureHandler)
+            withSecurityMatcher(
+                httpSecurity,
+                request ->
+                    isBasicAuthRequest(request)
+                        // If authentication isn't disabled for the API, all API requests without a
+                        // session cookie are treated as basic auth requests to work around a
+                        // limitation of Java's HTTP client, which only sends an Authorization
+                        // header after the server sends a WWW-Authenticate header in a 401
+                        // response.
+                        || !isApiUnprotected
+                            && isApiRequest(request)
+                            && !hasSessionCookie(request)),
+            authFailureHandler,
+            UNAUTHENTICATED_PATHS)
         .httpBasic(Customizer.withDefaults())
         .build();
   }
 
   @Bean
   @Profile("auth-basic")
   @Order(2)
-  public SecurityFilterChain loginAuthSecurityFilterChain(
-      final HttpSecurity httpSecurity, final AuthFailureHandler authFailureHandler)
+  public SecurityFilterChain unprotectedApiAccessSecurityFilterChain(
+      final HttpSecurity httpSecurity,
+      final AuthFailureHandler authFailureHandler,
+      final SecurityConfiguration securityConfiguration)
       throws Exception {
-    return buildLoginAuthSecurityFilterChain(httpSecurity, authFailureHandler);
+    if (!securityConfiguration.getAuthentication().getBasic().isApiDangerouslyUnprotected()) {
+      return null;
+    }
+    LOG.warn(
+        "The API is accessible without authentication. Remember to disable camunda.security.authentication.basic.api-dangerously-unprotected");
```

Comment:
👍  Nice that you make that transparent in the log! Just one additional suggestion
```suggestion
    LOG.warn(
        "The API is accessible without authentication. Remember to disable camunda.security.authentication.basic.api-dangerously-unprotected for a production server deployment.");
```

Reply:
It also should be disabled for a staging env, it's really only appropriate for local development. Maybe "for a non-local setup"?

Reply:
It's now 
> Please disable camunda.security.authentication.basic.api-dangerously-unprotected for any deployment.